beam-core/Database/Beam/Query.hs

```diff
@@ -361,8 +417,38 @@ updateTable tblEntity assignments mkWhere =
 
   in update tblEntity mkAssignments mkWhere
 
+-- | A specialization of 'update'' that is more convenient for normal tables.
+--
+--   Use 'updateTable' for comparisons with 'Bool'.
+updateTable' :: forall table db be
+              . ( BeamSqlBackend be, Beamable table )
+             => DatabaseEntity be db (TableEntity table)
+                -- ^ The table to update
+             -> table (QFieldAssignment be table)
+                -- ^ Updates to be made (use 'set' to construct an empty field)
+             -> (forall s. table (QExpr be s) -> QExpr be s SqlBool)
+             -> SqlUpdate be table
+updateTable' tblEntity assignments mkWhere =
+  let mkAssignments :: forall s. table (QField s) -> QAssignment be s
+      mkAssignments tblFields =
+        let tblExprs = changeBeamRep (\(Columnar' fd) -> Columnar' (current_ fd)) tblFields
+        in execWriter $
+           zipBeamFieldsM
+             (\(Columnar' field :: Columnar' (QField s) a)
+               c@(Columnar' (QFieldAssignment mkAssignment)) ->
+                case mkAssignment tblExprs of
+                  Nothing -> pure c
+                  Just newValue -> do
+                    tell (field <-. newValue)
+                    pure c)
+             tblFields assignments
+
+  in update' tblEntity mkAssignments mkWhere
```

Comment:
I honestly wasn't too concerned with sharing the implementation of these yet, since I was thinking we'd want to abstract a lot of these `Bool`/`SqlBool` pairs with a class eventually. However I see your point about type inference being dangerous. I was thinking we'd require explicit type at the "top level" like `guard_ @Bool $ x ==. y`, which would be even more clear than what we have now, but would be a huge breaking change.

Reply:
My general fear here is that the ideal solution isn't, to me, clear enough yet to actually have any sort of ETA. And if that ETA is, say, a year from now, then a lot of code could change before then. In that time I'd rather not risk fixes being done to primed versions of something and then not getting fixed for non-primed, or vice versa. But maybe my senses are wrong. It's not a strong opinion.

Reply:
Yeah. I definitely support abstracting out the implementation, and I agree we should avoid exporting it.